rotkehlchen/chain/ethereum/oracles/uniswap.py

```diff
@@ -280,6 +307,69 @@ def query_current_price(
         )
         return price, False
 
+    def _call_methods(
+            self,
+            pool_contract: EvmContract,
+            methods: list[str],
+            block_identifier: BlockIdentifier,
+            timestamp: Timestamp | None,
+    ) -> dict:
+        """Call methods on the pool contract using multicall if timestamp is None (for current)
+        or is after multicall contract creation, otherwise use individual calls.
+        Returns a dict that maps method names to responses.
+        """
+        if timestamp is None or timestamp > 1620277200:
+            response = self.ethereum.multicall(
+                calls=[
+                    (
+                        pool_contract.address,
+                        pool_contract.encode(method_name=method_name),
+                    )
+                    for method_name in methods
+                ],
+                require_success=True,
+                block_identifier=block_identifier,
+            )
+            output = {}
+            for index, method_name in enumerate(methods):
+                data = pool_contract.decode(response[index], method_name)
+                output[method_name] = data[0] if len(data) == 1 else data
+        else:
+            output = {
+                method_name: self.ethereum.call_contract(
+                    contract_address=pool_contract.address,
+                    abi=pool_contract.abi,
+                    method_name=method_name,
+                    block_identifier=block_identifier,
+                )
+                for method_name in methods
+            }
+        return output
+
+    def can_query_history(
+            self,
+            from_asset: Asset,
+            to_asset: Asset,
+            timestamp: Timestamp,
+            seconds: int | None = None,
+    ) -> bool:
```

Comment:
add a docstring stating that this is to comply with the interface for price historian